packages/transformers/src/modules/generic/http_post/http.post.transformer.ts

```diff
@@ -21,7 +23,23 @@ export class HttpPostTransformer implements ITransformer {
         this.config.headers = this.config.headers ?? xmsg.transformer?.metaData?.httpHeaders ?? {};
         this.config.headers = typeof this.config.headers === 'string' ? JSON.parse(this.config.headers || "{}") : this.config.headers ?? {};
         this.config.headers['Content-Type'] = 'application/json';
+        
         this.config.body = this.config.body ?? xmsg.transformer?.metaData?.httpBody ?? {};
+        const httpBodyCopy: Record<string, string> = {...this.config.body};
```

Comment:
```suggestion
        const httpBodyCopy: Record<string, string> = cloneDeep(this.config.body);
```

Should probably be a deep copy instead of a shallow copy.

Reply:
nope spread operator does not create a deep copy (made this mistake myself once :p)

Reply:
JavaScript gets weirder by the day.